grammar/test/GrammarTestRunner.java

```diff
@@ -0,0 +1,124 @@
+import org.antlr.v4.runtime.BailErrorStrategy;
+import org.antlr.v4.runtime.misc.ParseCancellationException;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+
+
+public class GrammarTestRunner {
+
+  GrammarTestRunner() {
+  }
+
+  // Track the number of tests run, and the success rate.
+  int error_count = 0;
+  int test_count = 0;
+
+  // Set to true for verbose diagnostic output.
+  boolean debug = false;
+
+  // Ensures an expression that is supposed to parse correctly does.
+  void testGood(String expression) {
+    if (!parseFloatingPointExpression(expression, debug)) {
+      error_count++;
+      System.out.println("Error! " + expression + " should succeed, but has a problem when parsing.");
+    }
+    test_count++;
+  }
+
+  // Ensures an expression that is _not_ supposed to parse correctly does not.
+  void testBad(String expression) {
+    if (parseFloatingPointExpression(expression, debug)) {
+      error_count++;
+      System.out.println("Error! " + expression + " should fail, but parses without issue.");
+    }
+    test_count++;
+  }
+
+  // Tries to parse with the "expression" rule, returning "False" if the parsing
+  // fails, and "True" if it succeeds.
+  static boolean parseFloatingPointExpression(String expr, boolean debug) {
+    CharStream stream = CharStreams.fromString(expr);
+
+    FloatingPointExpressionLexer lex = new FloatingPointExpressionLexer(stream);
+    CommonTokenStream tokens = new CommonTokenStream(lex);
+    FloatingPointExpressionParser parser = new FloatingPointExpressionParser(tokens);
+    // This prints diagnostic info about the parsing errors on the command line,
+    // which is not what we usually want, especially for cases we expect to fail
+    if (!debug)
+      parser.removeErrorListeners();
+    parser.setErrorHandler(new BailErrorStrategy());
+
+    try {
+      if (debug)
+        System.out.println("Parsing test expression: " + expr);
+      FloatingPointExpressionParser.ExpressionContext tree = parser.expression();
+      if (debug)
+        System.out.println("Parsed without exception");
+      return true;
+    } catch (ParseCancellationException ex) { //RecognitionException | NullPointerException | StringIndexOutOfBoundsException | RasterFormatException ex) {
```

Comment:
Remove the commented out exceptions here

Reply:
👍